fuel-vm/src/interpreter/gas.rs

```diff
@@ -21,37 +27,102 @@ impl<S, Tx> Interpreter<S, Tx> {
     }
 
     pub(crate) fn dependent_gas_charge(&mut self, gas_cost: DependentCost, arg: Word) -> Result<(), RuntimeError> {
-        if gas_cost.dep_per_unit == 0 {
-            self.gas_charge(gas_cost.base)
-        } else {
-            self.gas_charge(gas_cost.base.saturating_add(arg.saturating_div(gas_cost.dep_per_unit)))
-        }
+        let current_contract = self.contract_id();
+        let ReadRegisters { pc, ggas, cgas, is, .. } = split_registers(&mut self.registers).0;
+        let profiler = ProfileGas {
```

Comment:
does this add overhead when the profiler is disabled?

Reply:
What should I do instead though?

Reply:
`ProfileGas` can work with generic instead of `dyn`, and in this case, the compiler will optimize it. 

Also, you don't need at all to have a generic and a trait. You can have an alias like `type ProfileCapture = Profile;` when it is enabled and `type ProfileCapture = ProfileWithEmptyMethods;` when not, and use this alias in all places. And it seems you already do that, why not use an alias then without `dyn ProfileCapture`?

Reply:
Done in [this commit](https://github.com/FuelLabs/fuel-vm/pull/353/commits/bb4d87f04d3230362780ec57edce3facfc7311f6)